client/verta/tests/unit_tests/pipeline/test_registered_pipeline.py

```diff
@@ -0,0 +1,300 @@
+            graph=graph,
+            registered_model_version=make_mock_registered_model_version(),
+        )
+    copied_graph = pipeline.copy_graph()
+    # convert from sets to lists and sort for side-by-side comparison
+    graph_steps_sorted = sorted(list(graph.steps), key=lambda x: x.name)
+    copied_graph_steps_sorted = sorted(list(copied_graph.steps), key=lambda x: x.name)
+
+    for orig_step, copied_step in zip(graph_steps_sorted, copied_graph_steps_sorted):
+        assert orig_step is not copied_step
+        assert orig_step.name == copied_step.name
+        assert orig_step.predecessors == copied_step.predecessors
+        assert (
+            orig_step.registered_model_version.id
+            == copied_step.registered_model_version.id
+        )
+    assert copied_graph is not graph
+
+
+@given(model_version_name=st.text(min_size=1))
+@settings(
+    suppress_health_check=[HealthCheck.function_scoped_fixture],
+    deadline=None,
+)
+def test_log_pipeline_definition_artifact(
+    model_version_name,
+    mocked_responses,
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+) -> None:
+    """Verify the expected sequence of calls when a pipeline definition
+    is logged as an artifact to the pipeline's model version.
+
+    Fetching the registered model version is patched instead of mocking a
+    response to avoid having to pass the RM's id down through multiple
+    pytest fixtures.
+    """
+    rmv = make_mock_registered_model_version()
+    # Fetch the registered model version
+    mocked_responses.get(
+        f"{rmv._conn.scheme}://{rmv._conn.socket}/api/v1/registry/model_versions/{rmv.id}",
+        json={
+            "model_version": {
+                "id": rmv.id,
+                "registered_model_id": rmv.registered_model_id,
+                "version": model_version_name,
+            }
+        },
+        status=200,
+    )
+    mocked_responses.put(
+        f"{rmv._conn.scheme}://{rmv._conn.socket}/api/v1/registry/registered_models/{rmv.registered_model_id}/model_versions/{rmv.id}",
+        json={},
+        status=200,
+    )
+    # Fetch the artifact upload URL
+    mocked_responses.post(
+        f"{rmv._conn.scheme}://{rmv._conn.socket}/api/v1/registry/model_versions/{rmv.id}/getUrlForArtifact",
+        json={
+            "url": f"https://account.s3.amazonaws.com/development/ModelVersionEntity/"
+            f"{rmv.id}/pipeline.json"
+        },
+        status=200,
+    )
+    # Upload the artifact
+    mocked_responses.put(
+        f"https://account.s3.amazonaws.com/development/ModelVersionEntity/{rmv.id}/pipeline.json",
+        json={},
+        status=200,
+    )
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=rmv
+    ):
+        pipeline = RegisteredPipeline(
+            graph=make_mock_pipeline_graph(),
+            registered_model_version=rmv,
+        )
+    pipeline._log_pipeline_definition_artifact()
```

Comment:
Would you mind creating a ticket to have this test check the request body for correctness?

Reply:
Done